src/components/flame-graph/MaybeFlameGraph.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import explicitConnect from '../../utils/connect';
+import { getInvertCallstack } from '../../reducers/url-state';
+import { changeInvertCallstack } from '../../actions/profile-view';
+import FlameGraph from './FlameGraph';
+
+import type {
+  ExplicitConnectOptions,
+  ConnectedProps,
+} from '../../utils/connect';
+
+require('./MaybeFlameGraph.css');
+
+type StateProps = {|
+  +invertCallstack: boolean,
+|};
+type DispatchProps = {|
+  +changeInvertCallstack: typeof changeInvertCallstack,
+|};
+type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
```

Comment:
I put this connected component in its own file to avoid the problem of conflicting names, like for these types.

Reply:
For other components we changed the names to (for example) `<ComponentName>StateProps` to alleviate for the clashing names.

Reply:
That said, I think it's OK to have it in separate files :)